### PR TITLE
release notes for May 28th 2023

### PR DESCRIPTION
**Pull request by allyford**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+* After May 31, 2023, Ubuntu 18.04 will reach end of life. AKS will continue to update the host OS from Canonical into the Kubernetes 1.24 VHD images. Customers will not receive daily security updates from Canonical past the end of May, but will be able to consume those through a node image update only.
+* Each Kubernetes version is supported for 12 months. After 12 months, the minor version will shift to platform support only. Our new [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy) provides customers with Azure infrastructure support while the cluster is in an n-3 version (where n is the latest supported AKS GA minor version). Platform support does not include anything related to Kubernetes functionality and components, but provides customers with additional support beyond what was previously provided for unsupported versions.
+* Unattended Upgrades are disabled on Azure Linux when running on a NVIDIA GPU enabled VM sizes.
+* SecurityPatch OS Servicing channel is not supported on Azure Linux when running on NVIDIA GPU enabled VM sizes.
```
Comment (kaarthis) [inline CHANGELOG.md:13]: Link to the Security patch doc?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,54 @@
...
+  * [Overlay Network](https://azure.microsoft.com/en-us/updates/azurecnioverlay/) is now Generally available.
+  * Kubernetes Apps is now Generally available.
+
+* Preview Features
+  * [Generation 2 VMs](https://learn.microsoft.com/en-us/azure/aks/cluster-configuration#:~:text=Generation%202%20virtual%20machines%20on%20Windows%20(preview)) are now supported for Windows node pools. 
+  * [Custom Node Configuration for kubelet parameters](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration?tabs=linux-node-pools#:~:text=Prerequisites%20for%20Windows%20kubelet%20custom%20configuration%20(preview)) is now supported for Windows node pools.
+  * [Automated deployments](https://learn.microsoft.com/en-us/azure/aks/automated-deployments) now supports draft. Take your application and automatically create dockerfiles, kubernetes manifests, and github actions to deploy it onto your AKS cluster with ease.
+
+* Behavior Changes
+  * Customers may not upgrade to v1.25 and above if PSP is enabled, an error will occur if attempted.
```
Comment (kaarthis) [inline CHANGELOG.md:33]: What is the error , any GH link?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,54 @@
...
+  * Added hint to help guide users to install kubelogin if not already in their PATH.
+  * KMS v2 now supports clusters on kubernetes v1.27 and above. For kubernetes v1.27 and above, KMS v2 will be used as default for new clusters. For existing clusters upgraded to kubernetes 1.27 and above, KMS v1 will be used.
+  * Updated error handling for VMs failing with the same error code (50,51,52).
```
Comment (kaarthis) [inline CHANGELOG.md:36]: Again more information on the error would be helpful here.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,54 @@
...
+  * large-cluster-overrides.yaml is not disabled for free tier clusters in subscriptions included in the file.
+  * Web Application Routing now supports configuration through the Azure portal.
+   
+* Bug Fixes
```
Comment (kaarthis) [inline CHANGELOG.md:40]: Wherever possible a GH link like the one you have in second item is helpful.

**Review by kaarthis — CHANGES_REQUESTED** [review]
Kindly make these changes.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,54 @@
...
+
+### Release notes
+
+* Features
+  * Azure Linux is now generally available as a container host OS on AKS. The Build announcement can be found [here](https://build.microsoft.com/en-US/sessions/379005cd-6a55-4511-8d43-c5ceaad34e60?source=sessions) and the documentation for deploying Azure Linux can be found [here](https://learn.microsoft.com/en-us/azure/azure-linux/intro-azure-linux).
+  * FIPS image support is now enabled for Azure Linux.
+  * The [AKS devX extension](https://github.com/Azure/aks-devx-tools) now supports the creation of GitHub Actions.
+  * Azure AD [Workload Identity](https://aka.ms/aks/workloadidentity) is now Generally available.
+  * Managed Prometheus and Grafana are now Generally available.
+  * [Overlay Network](https://azure.microsoft.com/en-us/updates/azurecnioverlay/) is now Generally available.
```
Comment (palma21) [inline CHANGELOG.md:24]: Overlay was GAed at kubecon. Is this about cilium @phealy @chasewilson ?

Comment (allyford) [reply CHANGELOG.md:24]: But the link there is from April so fine to remove since already announced

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,53 @@
...
+* Each Kubernetes version is supported for 12 months. After 12 months, the minor version will shift to platform support only. Our new [platform support policy](https://learn.microsoft.com/azure/aks/supported-kubernetes-versions?tabs=azure-cli#platform-support-policy) provides customers with Azure infrastructure support while the cluster is in an n-3 version (where n is the latest supported AKS GA minor version). Platform support does not include anything related to Kubernetes functionality and components, but provides customers with additional support beyond what was previously provided for unsupported versions.
+* Unattended Upgrades are disabled on Azure Linux when running on a NVIDIA GPU enabled VM sizes.
+* [SecurityPatch OS Servicing channel](https://learn.microsoft.com/en-us/azure/aks/auto-upgrade-node-image) is not supported on Azure Linux when running on NVIDIA GPU enabled VM sizes.
+* Windows2019 will be retired in Kubernetes v1.33 and above (ETA March 2026). Customers should [upgrade to Windows2022](https://learn.microsoft.com/en-us/azure/aks/upgrade-windows-2019-2022).
```
Comment (shashankbarsin) [inline CHANGELOG.md:14]:
```suggestion
* Windows2019 will be retired in Kubernetes v1.33 and above (ETA March 2026). Customers should [upgrade to Windows2022](https://learn.microsoft.com/azure/aks/upgrade-windows-2019-2022).
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,53 @@
...
+
+### Release notes
+
+* Features
+  * Azure Linux is now generally available as a container host OS on AKS. The Build announcement can be found [here](https://build.microsoft.com/en-US/sessions/379005cd-6a55-4511-8d43-c5ceaad34e60?source=sessions) and the documentation for deploying Azure Linux can be found [here](https://learn.microsoft.com/en-us/azure/azure-linux/intro-azure-linux).
```
Comment (shashankbarsin) [inline CHANGELOG.md:19]:
```suggestion
  * Azure Linux is now generally available as a container host OS on AKS. The Build announcement can be found [here](https://build.microsoft.com/en-US/sessions/379005cd-6a55-4511-8d43-c5ceaad34e60?source=sessions) and the documentation for deploying Azure Linux can be found [here](https://learn.microsoft.com/azure/azure-linux/intro-azure-linux).
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,53 @@
...
+  * FIPS image support is now enabled for Azure Linux.
+  * The [AKS devX extension](https://github.com/Azure/aks-devx-tools) now supports the creation of GitHub Actions.
```
Comment (shashankbarsin) [inline CHANGELOG.md:21]: @qpetraroia - new action itself or is it actions workflow?

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,53 @@
...
+  * Azure AD [Workload Identity](https://aka.ms/aks/workloadidentity) is now Generally available.
+  * [Managed Prometheus](https://learn.microsoft.com/en-Us/azure/azure-monitor/essentials/prometheus-metrics-overview) and [Grafana](https://learn.microsoft.com/en-us/azure/managed-grafana/overview) are now Generally available.
```
Comment (shashankbarsin) [inline CHANGELOG.md:23]: we need to remove en-us. and it's only managed prometheus that's newly GA right? Managed Grafana GA was a while ago iirc?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,53 @@
...
+  * [Kubernetes Apps](https://aka.ms/deployK8sApp) is now Generally available.
+
+* Preview Features
+  * [Generation 2 VMs](https://learn.microsoft.com/en-us/azure/aks/cluster-configuration#:~:text=Generation%202%20virtual%20machines%20on%20Windows%20(preview)) are now supported for Windows node pools. 
+  * [Custom Node Configuration for kubelet parameters](https://learn.microsoft.com/en-us/azure/aks/custom-node-configuration?tabs=linux-node-pools#:~:text=Prerequisites%20for%20Windows%20kubelet%20custom%20configuration%20(preview)) is now supported for Windows node pools.
```
Comment (shashankbarsin) [inline CHANGELOG.md:28]:
```suggestion
  * [Custom Node Configuration for kubelet parameters](https://learn.microsoft.com/azure/aks/custom-node-configuration?tabs=linux-node-pools#:~:text=Prerequisites%20for%20Windows%20kubelet%20custom%20configuration%20(preview)) is now supported for Windows node pools.
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,53 @@
...
+  * [Automated deployments](https://learn.microsoft.com/en-us/azure/aks/automated-deployments) now supports draft. Take your application and automatically create dockerfiles, kubernetes manifests, and github actions to deploy it onto your AKS cluster with ease.
```
Comment (shashankbarsin) [inline CHANGELOG.md:29]:
```suggestion
  * [Automated deployments](https://learn.microsoft.com/azure/aks/automated-deployments) now supports draft. Take your application and automatically create dockerfiles, kubernetes manifests, and github actions to deploy it onto your AKS cluster with ease.
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,53 @@
...
+
+* Behavior Changes
+  * Customers may not upgrade to v1.25 and above if PSP is enabled, an error will occur if attempted.
+  * Added hint to help guide users to install kubelogin if not already in their PATH.
```
Comment (shashankbarsin) [inline CHANGELOG.md:33]: hint in? az aks install-cli? any doc link?

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,53 @@
...
+  * KMS v2 now supports clusters on kubernetes v1.27 and above. For kubernetes v1.27 and above, KMS v2 will be used as default for new clusters. For existing clusters upgraded to kubernetes 1.27 and above, KMS v1 will be used.
```
Comment (shashankbarsin) [inline CHANGELOG.md:34]: has 1.27 been released? didn't see it in the notes here

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,53 @@
...
+  * Updated error handling for VMs failing with the same error code (50,51,52).
+  * large-cluster-overrides.yaml is not disabled for free tier clusters in subscriptions included in the file.
```
Comment (shashankbarsin) [inline CHANGELOG.md:36]: this may be internal only. if yes, needs to be removed

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,53 @@
...
+  * Web Application Routing now supports configuration through the Azure portal.
+   
+* Bug Fixes
+  * Fixed a bug to make windows azure-cns work in kubernetes v1.27 and above.
```
Comment (shashankbarsin) [inline CHANGELOG.md:40]: same as above. need to include when we have 1.27 available

Comment (qpetraroia) [reply CHANGELOG.md:39]: Would probably just remove this as 1.27 isn't out yet

**Review by qpetraroia — APPROVED** [review]

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,50 @@
...
+* [SecurityPatch OS Servicing channel](https://learn.microsoft.com/azure/aks/auto-upgrade-node-image) is not supported on Azure Linux when running on NVIDIA GPU enabled VM sizes.
+* Windows2019 will be retired in Kubernetes v1.33 and above (ETA March 2026). Customers should [upgrade to Windows2022](https://learn.microsoft.com/azure/aks/upgrade-windows-2019-2022).
+
+### Release notes
+
+* Features
+  * Azure Linux is now generally available as a container host OS on AKS. The Build announcement can be found [here](https://build.microsoft.com/sessions/379005cd-6a55-4511-8d43-c5ceaad34e60?source=sessions) and the documentation for deploying Azure Linux can be found [here](https://learn.microsoft.com/azure/azure-linux/intro-azure-linux).
+  * FIPS image support is now enabled for Azure Linux.
+  * The [AKS devX extension](https://github.com/Azure/aks-devx-tools) now supports the creation of GitHub Actions.
+  * [Managed Prometheus](https://learn.microsoft.com/azure/azure-monitor/essentials/prometheus-metrics-overview) and [Grafana](https://learn.microsoft.com/azure/managed-grafana/overview) are now Generally available.
```
Comment (shashankbarsin) [inline CHANGELOG.md:22]: checked - managed grafana GA was announced end of Aug. only the managed prometheus GA announcement is new. we can remove the grafana part

File: CHANGELOG.md (hunk 8 above)
Comment (qpetraroia) [reply CHANGELOG.md:21]: regular gh actions @shashankbarsin. We added a command to allow you to create them in the devX extension

**Review by kaarthis — APPROVED** [review]
Approving with a small nit on the labels point.

**Pull request merged**